core/metrics-core-service/src/main/java/org/hawkular/metrics/core/jobs/JobsServiceImpl.java

```diff
@@ -102,8 +104,10 @@ public void start() {
         if(compressionEnabled) {
             CompressData compressDataJob = new CompressData(metricsService);
             scheduler.register(CompressData.JOB_NAME, compressDataJob);
-            maybeScheduleCompressData();
+            maybeScheduleCompressData(backgroundJobs);
```

Comment:
Wouldn't it make sense to have this method do the insert to the list and the scheduling parts just return the jobDetails?